Dockerfile

```diff
@@ -1,35 +1,51 @@
-ARG NODE_VERSION=10.15.3
+# Set Node version
+FROM node:10.15.3-alpine AS node-base
+
+# Production base
+FROM node-base AS production-base
```

Comment:
The key thing is that we need to use dev dependencies to compile Sass but don't want to include dev dependencies themselves in any stage of the production build, since that would greatly increase the production image size. We also don't want to download production dependencies twice, so we added a production-base stage to fetch those and copy them across to each of the dev and production stages.

The COPY step got missed from the development stage so we're actually not getting the benefit of this yet. Additionally, I've now realised it would be more efficient for the dev and prod stages to each extend production-base, rather than adding a COPY step to each.

Reply:
nice, the new structure looks to be the most efficient.